contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -3950,26 +3957,29 @@ load_categories_hash(RawStmt *cats_sql, MemoryContext per_query_ctx)
 					 errmsg("provided \"categories\" SQL must " \
 							"return 1 column of at least one row")));
 
+		Assert(spi_tuptable->numvals >= proc);
 		for (i = 0; i < proc; i++)
 		{
 			bbf_pivot_cat_desc *catdesc;
 			char	   *catname;
+			char	   *catname_lower;
 			HeapTuple	spi_tuple;
 
 			/* get the next sql result tuple */
 			spi_tuple = spi_tuptable->vals[i];
 
 			/* get the category from the current sql result tuple */
 			catname = SPI_getvalue(spi_tuple, spi_tupdesc, 1);
-			if (catname == NULL)
+			catname_lower = downcase_identifier(catname, strlen(catname), false, false);
+			if (catname_lower == NULL)
 				ereport(ERROR,
 						(errcode(ERRCODE_SYNTAX_ERROR),
 						 errmsg("provided \"categories\" SQL must " \
 								"not return NULL values")));
 			
 			SPIcontext = MemoryContextSwitchTo(per_query_ctx);
 			catdesc = (bbf_pivot_cat_desc *) palloc(sizeof(bbf_pivot_cat_desc));
-			catdesc->catname = catname;
+			catdesc->catname = pstrdup(catname_lower);
 			catdesc->attidx = i;
 			/* Add the proc description block to the hashtable */
```

Comment:
I think it better use tuple description block in the comment

proc in PG context normally mean procedure.

Reply:
yes, will change the var name